### DISCUSSION TITLE
Midi devices not detected after failed upgrade and revert

### DISCUSSION THREAD

**Discussion by OrgelSpieler** [GrandOrgue/grandorgue#1798] [Help]
I've been running GO 3.9.5-1 on a Win7 platform with dual Xeon processors, 48 GB of memory, and an RME HDSPe AIO sound card with no issues for a long time.  This evening I made an attempt to upgrade to the latest GO version, but it failed miserably.  I used Windows Restore to revert to my previous state,  It was successful with one problem.  I have sound output and everything works as before, but the Midi Settings panel shows no midi REM device (see photo).  (The RME device was previously showing as I reviewed all the settings before I attempted the upgrade).  If I check the Midi settings for any manual, it shows the RME device is present (see photo).  I tried a GO re-install, but it did not change the outcome.  While I can use my current organ sample set in this state, it is preventing me from installing a new sample set.   The RME device is showing in Device Manager and has the latest driver.  I thought I need to resolve this before I attempt the upgrade again.  Any ideas in how to fix this are appreciated.  Thanks.

![GO Midi Devices](https://github.com/GrandOrgue/grandorgue/assets/22777068/a879f9d4-75d9-4f30-8a22-7ca983ea5e8f)
![GO Midi Manual Settings](https://github.com/GrandOrgue/grandorgue/assets/22777068/0c192200-0355-4863-a33a-7a483885e768)

**Comment by oleg68** [comment]
Seems Windows Restore broke the MIDI layer, so the operating system doesn't more expose your midi device to GrandOrgue.

Please try to install [MidiOx](http://www.midiox.com/) and test whether your device is present in MidiOx? If no, you need to reinstall the RME driver or even to reinstall Windows. If the device is present there, tell me it, and I give you an executable with additional debug.

For the future: if you want to try a new GrandOrgue version, you can do it without installation: just download the .zip, unpack it to any directory and run it from this directory. Later, if you like it, you may to install the new version otherwise you can just delete the folder without necessarity of Windows Restore.

**Reply by OrgelSpieler** [reply]
MidiOx showed no midi devices present.  I re-installed the RME driver, but that did not resolve the issue.  Rather than re-install Windows,  I tried a different restore point which successfully returned everything to normal.  Had that failed, I would have a attempted a full restore from a nightly backup copy before doing a Windows re-install.
Thanks for the MidiOx tip.  I would not have initially suspected a Windows midi problem because the device was showing in Device manager and I had the ability to play the last organ loaded in GO.   Thanks for pointing me in the right direction and the quick response.
